File: internal/core/run/checks.go
// Package run contains shared domain types for Terraform run and check results.
package run

import "slices"

// CheckResult represents a check result from a Terraform plan or state.
type CheckResult struct {
	Name    string
	Status  string
	Objects []CheckResultObject
}

// CheckResultObject represents an individual checkable object instance within a check result.
type CheckResultObject struct {
	Address         string
	Status          string
	FailureMessages []string
}

var validCheckStatuses = []string{"pass", "fail", "error", "unknown"}

// NormalizeCheckStatus returns the status unchanged if it is a known Terraform check
// status value, or "unknown" for any unrecognized input.
func NormalizeCheckStatus(status string) string {
	if slices.Contains(validCheckStatuses, status) {
		return status
	}
	return "unknown"
}

File: frontend/src/workspace/checks/checks.ts
// Shared helpers for rendering Terraform check result statuses.

export function getCheckStatusLabel(status: string): string {
    switch (status) {
        case 'PASS': return 'Pass';
        case 'FAIL': return 'Fail';
        case 'ERROR': return 'Error';
        case 'UNKNOWN': return 'Unknown';
        default: return status;
    }
}

export function getCheckStatusTooltip(status: string): string {
    switch (status) {
        case 'PASS': return 'The check assertion condition evaluated to true';
        case 'FAIL': return 'The check assertion condition evaluated to false';
        case 'ERROR': return 'Terraform could not evaluate the check condition';
        case 'UNKNOWN': return 'The check result could not be determined';
        default: return '';
    }
}

export function collectFailureMessages(objects: readonly { readonly failureMessages: readonly string[] }[]): readonly string[] {
    return objects.flatMap((obj) => obj.failureMessages);
}
